integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -642,6 +655,71 @@ def test_hash_groupby_collect_set_on_nested_type(data_gen):
             .groupby('a')
             .agg(f.sort_array(f.collect_set('b'))))
 
+
+# Note, using sort_array() on the CPU, because sort_array() does not yet
+# support sorting certain nested/arbitrary types on the GPU
+# See https://github.com/NVIDIA/spark-rapids/issues/3715
+# and https://github.com/rapidsai/cudf/issues/11222
+@approximate_float
+@ignore_order(local=True)
+@incompat
+@allow_non_gpu("ProjectExec", "SortArray")
+@pytest.mark.parametrize('data_gen', _gen_data_for_collect_set_op_nested, ids=idfn)
+def test_hash_groupby_collect_set_on_nested_array_type(data_gen):
+    conf = {
+        "spark.rapids.sql.castFloatToString.enabled": "true",
+        "spark.rapids.sql.castDecimalToString.enabled": "true",
+        "spark.rapids.sql.expression.SortArray": "false"
+    }
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, data_gen, length=100)
+            .groupby('a')
+            .agg(f.collect_set('b').alias("collect_set"))
+            .selectExpr("sort_array(collect_set)")
+        , conf=conf)
+
+@approximate_float
```

Comment:
Why would collect_set need approximate_float? We should not be changing any values? This goes for all of the tests on `collect_set`.  Looks like the `collect_list` ones do not have the same tag.

Reply:
Should not need that. `collect_set` in cudf is implemented by gathering the input distinct rows thus the output should be exact, not approximate.